moveit_ros/visualization/planning_scene_rviz_plugin/src/planning_scene_display.cpp

```diff
@@ -462,36 +470,33 @@ void PlanningSceneDisplay::unsetLinkColor(const std::string& link_name)
     unsetLinkColor(&planning_scene_robot_->getRobot(), link_name);
 }
 
-void PlanningSceneDisplay::setLinkColor(rviz::Robot* robot, const std::string& link_name, const QColor& color)
+void PlanningSceneDisplay::setLinkColor(rviz_default_plugins::robot::Robot* robot, const std::string& link_name,
+                                        const QColor& color)
 {
-  rviz::RobotLink* link = robot->getLink(link_name);
+  rviz_default_plugins::robot::RobotLink* link = robot->getLink(link_name);
 
   // Check if link exists
   if (link)
     link->setColor(color.redF(), color.greenF(), color.blueF());
 }
 
-void PlanningSceneDisplay::unsetLinkColor(rviz::Robot* robot, const std::string& link_name)
+void PlanningSceneDisplay::unsetLinkColor(rviz_default_plugins::robot::Robot* robot, const std::string& link_name)
 {
-  rviz::RobotLink* link = robot->getLink(link_name);
+  rviz_default_plugins::robot::RobotLink* link = robot->getLink(link_name);
 
   // Check if link exists
   if (link)
     link->unsetColor();
 }
-
 // ******************************************************************************************
 // Load
 // ******************************************************************************************
-planning_scene_monitor::PlanningSceneMonitorPtr PlanningSceneDisplay::createPlanningSceneMonitor()
+planning_scene_monitor::PlanningSceneMonitorPtr
+PlanningSceneDisplay::createPlanningSceneMonitor(const rclcpp::Node::SharedPtr node)
 {
-#ifdef RVIZ_TF1
   std::shared_ptr<tf2_ros::Buffer> tf_buffer = moveit::planning_interface::getSharedTF();
```

Comment:
Shouldn't we get the TF2 buffer here?

Reply:
Unfortunately, there's no function that returns a tf2 buffer this's why I chose the old way